server/app/repository/ExportServiceRepository.java

```diff
@@ -0,0 +1,93 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Provider;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.inject.Inject;
+import models.Version;
+import services.question.types.MultiOptionQuestionDefinition;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionType;
+
+/** Implements queries related to CSV exporting needs. */
+public final class ExportServiceRepository {
+  private final Database database;
+  private final Provider<VersionRepository> versionRepositoryProvider;
+
+  @Inject
+  public ExportServiceRepository(Provider<VersionRepository> versionRepositoryProvider) {
+    this.database = DB.getDefault();
+    this.versionRepositoryProvider = checkNotNull(versionRepositoryProvider);
+  }
+
+  /**
+   * This method creates a map of all the optionIds to its option's adminName string for a given
+   * multiselect question definition There are two queries which are run here 1. To get all the
+   * adminNames to optionIds for a given question 2. To get all the options which were ever picked
+   * by an applicant (this is to filter out options accidentally introduced) The options are now
+   * combined with the latest active version of the question and returned as an immutable map. It
+   * throws RuntimeException when the question is not of type Checkbox It also throws
+   * NoSuchElementException when the active version of the question is missing
+   */
```

Comment:
**style** and **typos**: This description is very detailed and helpful! A few style and formatting notes:
- Adding a summary fragment would be helpful: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment
- Adding line-breaks for each item in the list
- Adding periods after the sentences
- Using the `{@link RuntimeException}` formatting so the javadoc generates links. Like https://github.com/civiform/civiform/blob/3440253cd3abeeda48abe3c072d2ada8efd979b0/server/app/services/Path.java#L129

Try mousing over the method call and see if the javadoc IntelliJ generates is formatted in a useful way.